File: community/cypher/cypher-compiler-2.3/src/test/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/codegen/SetStaticFieldTest.scala
package org.neo4j.cypher.internal.compiler.v2_3.birk.codegen

import org.neo4j.cypher.internal.compiler.v2_3.test_helpers.CypherFunSuite

class SetStaticFieldTest extends CypherFunSuite {
  test("do it") {
    setStaticField(classOf[APA], "X", "HELLO WORLD!")

    APA.X should equal("HELLO WORLD!")
  }
}

object APA {
  var X : String = ""
  val Xval : String = ""
  def Xdef : String = ""
}

class APA

File: community/cypher/cypher-compiler-2.3/src/test/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/codegen/SetStaticFieldTest.java
package org.neo4j.cypher.internal.compiler.v2_3.birk.codegen;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class SetStaticFieldTest
{
    @Test
    public void shouldAssignFields()
    {
        // when
        setStaticField.apply( Apa.class, "X", "HELLO WORLD!" );

        // then
        assertEquals( Apa.X, "HELLO WORLD!" );
    }

    public static class Apa
    {
        public static String X;
    }
}